src/spiral_stc.cpp

```diff
@@ -8,294 +8,356 @@
   }
-  return pathNodes;
-}
 
-std::list<Point_t> SpiralSTC::spiral_stc(std::vector<std::vector<bool> > const& grid,
-                                          Point_t& init,
-                                          int &multiple_pass_counter,
-                                          int &visited_counter)
-{
-  int x, y;
-  // Initial node is initially set as visited so it does not count
-  multiple_pass_counter = 0;
-  visited_counter = 0;
-
-  std::vector<std::vector<bool> > visited;
-  visited = grid;  // Copy grid matrix
-  x = init.x;
-  y = init.y;
-
-  Point_t new_point = { x, y };
-  gridNode_t new_node =
+  std::list<Point_t> SpiralSTC::spiral_stc(std::vector<std::vector<bool>> const &grid,
+                                           Point_t &init,
+                                           int &multiple_pass_counter,
+                                           int &visited_counter)
   {
-    new_point,  // Point: x,y
-    0,          // Cost
-    0,          // Heuristic
-  };
-  std::list<gridNode_t> pathNodes;
-  std::list<Point_t> fullPath;
-  pathNodes.push_back(new_node);
-  visited[y][x] = eNodeVisited;
+    int x, y;
+    // Initial node is initially set as visited so it does not count
+    multiple_pass_counter = 0;
+    visited_counter = 0;
+
+    std::vector<std::vector<bool>> visited;
+    visited = grid; // Copy grid matrix
+    x = init.x;
+    y = init.y;
+
+    Point_t new_point = {x, y};
+    gridNode_t new_node =
+        {
+            new_point, // Point: x,y
+            0,         // Cost
+            0,         // Heuristic
+        };
+    std::list<gridNode_t> pathNodes;
+    std::list<Point_t> fullPath;
+    pathNodes.push_back(new_node);
+    visited[y][x] = eNodeVisited;
 
 #ifdef DEBUG_PLOT
-  RCLCPP_INFO(rclcpp::get_logger("FullCoveragePathPlanner"), "Grid before walking is: ");
-  printGrid(grid, visited, fullPath);
+    RCLCPP_INFO(rclcpp::get_logger("FullCoveragePathPlanner"), "Grid before walking is: ");
+    printGrid(grid, visited, fullPath);
 #endif
 
-  pathNodes = SpiralSTC::spiral(grid, pathNodes, visited);                // First spiral fill
-  std::list<Point_t> goals = map_2_goals(visited, eNodeOpen);  // Retrieve remaining goalpoints
-  // Add points to full path
-  std::list<gridNode_t>::iterator it;
-  for (it = pathNodes.begin(); it != pathNodes.end(); ++it)
-  {
-    Point_t newPoint = { it->pos.x, it->pos.y };
-    visited_counter++;
-    fullPath.push_back(newPoint);
-  }
-  // Remove all elements from pathNodes list except last element
-  pathNodes.erase(pathNodes.begin(), --(pathNodes.end()));
+    pathNodes = SpiralSTC::spiral(grid, pathNodes, visited);    // First spiral fill
+    std::list<Point_t> goals = map_2_goals(visited, eNodeOpen); // Retrieve remaining goalpoints
+    // Add points to full path
+    std::list<gridNode_t>::iterator it;
+    for (it = pathNodes.begin(); it != pathNodes.end(); ++it)
+    {
+      Point_t newPoint = {it->pos.x, it->pos.y};
```

Comment:
Would you mind converting these to auto-loops? So I don't have to figure out in the review whether you included the first and/or the last index as well.

```suggestion
    for (const auto pathNode : pathNodes)
    {
      Point_t newPoint = {pathNode.x, pathNode.y};
```